NikolayGrabenko/src/library/controller/Library.java

```diff
@@ -0,0 +1,298 @@
+package library.controller;
+
+import library.model.Editions;
+import library.model.PrintedEditions;
+import library.model.Reader;
+import library.utils.ArrayListMy;
+
+import java.util.Comparator;
+
+public class Library {
+
+    Reader reader;
+    Editions edition;
+
+    private ArrayListMy printedEditionsBase;
+    private ArrayListMy printedEditionsAvailable;
+    private ArrayListMy printedEditionsAtReaders;
+
+    private ArrayListMy registeredReaders;
+    private ArrayListMy blackListOfReaders;
+
+
+    public ArrayListMy getPrintedEditionsBase() {
+        return printedEditionsBase;
+    }
+
+    public ArrayListMy getPrintedEditionsAtReaders() {
+        return printedEditionsAtReaders;
+    }
+
+    public ArrayListMy getPrintedEditionsAvailable() {
+        return printedEditionsAvailable;
+    }
+
+    public ArrayListMy getRegisteredReaders() {
+        return registeredReaders;
+    }
+
+    public ArrayListMy getBlackListOfReaders() {
+        return blackListOfReaders;
+    }
+
+    public Library() {
+        printedEditionsBase = new ArrayListMy();
+        printedEditionsAvailable = new ArrayListMy();
+        printedEditionsAtReaders = new ArrayListMy();
+        registeredReaders = new ArrayListMy();
+        blackListOfReaders = new ArrayListMy();
+
+    }
+
+    public boolean isRegistered(Reader someReader) {
+        boolean isRegistered = false;
+        for (int i = 0; i < registeredReaders.size(); i++) {
+            if (registeredReaders.get(i).equals(someReader)) {
+                isRegistered = true;
+                break;
+            }
+        }
+        return isRegistered;
+    }
+
+    public boolean isInBlackList(Reader reader) {
+        boolean isInBlackList = false;
+        if (reader != null) {
+            for (int i = 0; i < blackListOfReaders.size(); i++) {
+                Reader readerTemp = (Reader) blackListOfReaders.get(i);
+                if (readerTemp.equals(reader)) {
+                    isInBlackList = true;
+                    break;
+                }
+            }
+        }
+        return isInBlackList;
+    }
+
+    public boolean isAvailablePrintEds(PrintedEditions edition) {
+        boolean isAvailablePrintEds = false;
+        if (edition != null) {
+            for (int i = 0; i < printedEditionsAvailable.size(); i++) {
+                PrintedEditions editionTemp = (PrintedEditions) printedEditionsAvailable.get(i);
+                if (editionTemp.getNumberOfCopiesAvailable() > 0) {
+                    isAvailablePrintEds = true;
+                    break;
+                }
+            }
+        }
+        return isAvailablePrintEds;
+    }
+    //    HELP!!!!!
+    public void sortPrintedEditionByComparator(ArrayListMy list, Comparator comparator) {
```

Comment:
public void sortPrintedEditionByComparator(ArrayListMy list, Comparator comparator) {
        for (int i = 0; i < list.size(); i++) {
            for (int j = list.size() - 1; j > i; j--) {
                    if (comparator.compare(list.get(j), list.get(j - 1)) < 0) {
                        PrintedEditions editionTemp = (PrintedEditions) list.get(j);
                        list.set(j, list.get(j - 1));
                        list.set((j - 1), editionTemp);
                }
            }
        }
    }